### PR TITLE
Update grid

### PR DESCRIPTION
**Pull request by ahulist**
In some cases (e.g. use of [np.memmap](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html)) it is time-wise advantageous to swap underlying dense data by accessing private `_memory_blocks` and `_grid_mask`. In such case `_grid` keys remain out of date. Private method in this PR allows for `_grid` regeneration.

### CODE REVIEW

File: chunky3d/chunky.py [hunk 1]
```diff
@@ -495,12 +497,14 @@ def update_grid_mask(self):
...
+                self._block_shape, Sparse.EMPTY_GRID_VALUE, dtype=np.uint32
+            )
         else:
-            self._grid_mask.fill(0xFFFFFFFF)
+            self._grid_mask.fill(Sparse.EMPTY_GRID_VALUE)
 
-        self._grid_mask[tuple(np.array(sorted(self._grid.keys())).T)] = np.arange(
-            self.nchunks_initialized
+        self._grid_mask[tuple(np.array(sorted(self._grid.keys())).T)] = (
+            np.arange(self.nchunks_initialized)
```
Comment (tgandor) [inline chunky3d/chunky.py]: Is it `black`?

File: chunky3d/chunky.py [hunk 2]
```diff
@@ -567,6 +571,26 @@ def _update_memory_block(self, idx, d):
         else:
             self._grid[idx][:] = d
 
+    def _update_grid(self):
+        """In case the underlying `_memory_blocks` and `_grid_mask` change, this method updates the `_grid`."""
+        chunks = self.chunks
+        grid = {}
+        for i, j, k in np.argwhere(self.grid_mask != Sparse.EMPTY_GRID_VALUE):
+            idx = self._grid_mask[i, j, k]
+            envelope = 0
```
Comment (tgandor) [inline chunky3d/chunky.py:583]: Is this intentional? The access to `dense_data` below would be simplified if this variable disappeared.
Or was is supposed to be != 0 in special cases?

File: chunky3d/chunky.py (hunk 1 above)
Comment (ahulist) [reply chunky3d/chunky.py]: It was, I just had different line length settings. Reverted the change.

File: chunky3d/chunky.py (hunk 2 above)
Comment (ahulist) [reply chunky3d/chunky.py:583]: Added the missing parameter.

Comment (tgandor) [reply chunky3d/chunky.py:583]: OK, but now this assignment needs to be removed.

**Pull request merged**